ibis_heavyai/operations.py

```diff
@@ -804,7 +804,11 @@ def _arbitrary(translator, expr):
 class NumericTruncate(ops.NumericBinaryOp):
     """Truncates x to y decimal places."""
 
-    output_type = rlz.shape_like('left', dt.float)
+    try:
+        output_dtype = rlz.dtype_like('left')
+        output_shape = rlz.shape_like('left')
+    except AttributeError:  # ibis<3
+        output_type = rlz.shape_like('left', dt.float)
```

Comment:
Could we explicitly test ibis version instead of using try-except construct (that typically makes discovering bugs harder)? E.g.
```
if ibis_version[0] < 3:
    output_type = rlz.shape_like('left', dt.float)
else:
    output_dtype = rlz.dtype_like('left')
    output_shape = rlz.shape_like('left')
```
Same comment applies to other places.

Reply:
I will do it properly with the packaging module then.